### ISSUE TITLE
Remove nn.module

### ISSUE THREAD

**Issue by govindansriram** [govindansriram/CobraML2#25]
`nn.Module` is the class all layers are built on. It provides a clean interface that supports gradient tracking, forward passes, and more. When doing inference, this stuff doesn't matter, and it just adds overhead. We should mimic `mini-sglang` and create our own layer type that contains the weights, and the computation thats it.

https://github.com/sgl-project/mini-sglang/blob/82722ad6dc85df766278c48061d768b4117a3bd4/python/minisgl/layers/base.py#L16-L82

This requires a lot of work, since it not only affects the layers but also how we load the state dict back into the layer.

Depends on https://github.com/govindansriram/CobraML2/issues/23
